apps/os/backend/agent/agent-core.ts

```diff
@@ -1316,9 +1316,79 @@ export class AgentCore<
       case "CORE:INTERNAL_ERROR":
       case "CORE:INITIALIZED_WITH_EVENTS":
       case "CORE:LOG":
-      case "CORE:BACKGROUND_TASK_PROGRESS":
-        // Just log, no state change needed
+      case "CORE:BACKGROUND_TASK_PROGRESS": {
+        // Maintain one developer message per processId with COMPLETE aggregated stdout/stderr
+        const {
+          processId,
+          stdout: latestStdout = "",
+          stderr: latestStderr = "",
+          lastSeq,
+          complete,
+        } = event.data as {
```

Comment:
why do we need to cast? shouldn't we define the types in agent-core-schemas.ts such that this can be inferred?